test/runtests.jl

```diff
@@ -1,7 +1,24 @@
 using Intervals
 using TimeZones
-using Base.Test
-using Base.Dates
+using Compat
+using Compat.Test
+using Compat.Dates
+
+# The method based on keyword arguments is not in Compat, so to avoid
+# deprecation warnings on 0.7 we need this little definition.
+if VERSION >= v"0.7.0-DEV.4524"
+    compactstring(x) = sprint(show, x, context=:compact=>true)
+else
+    compactstring(x) = sprint(showcompact, x)
+end
```

Comment:
Although this won't work generally in `Compat` we can use this here to use the new 0.7 `sprint` syntax instead of making a new function:
```julia
if VERSION < v"0.7.0-DEV.4524"
    function sprint(f::Function, args...; context=nothing)
        if context !== nothing
            Base.sprint((io, args...) -> f(IOContext(io, context), args...), args...)
        else
            Base.sprint(f, args...)
        end
    end
end
```
Then we can just use `sprint(show, x, context=:compact=>true)` in the tests.